app/helpers/katello/subscription_mailer_helper.rb

```diff
@@ -0,0 +1,18 @@
+module Katello
+  module SubscriptionMailerHelper
+    include Rails.application.routes.url_helpers
+
+    def report_url
+      base_url = report_data_report_template_path(@report_template.id, job_id: @provider_job_id) unless @pools.empty?
+      "#{Setting[:foreman_url]}#{base_url}"
+    end
+
+    def start_report_task(mail_query)
+      @report_template ||= ReportTemplate.find_by(name: "Subscription - Entitlement Report")
+      template_input_id = TemplateInput.find_by_name("Days from Now").id.to_s
```

Comment:
I just learned something. TemplateInput 'names' are not unique - so I think we need to make sure that we are looking at the TemplateInput with the name 'Days from Now' that belongs to our report template. think 'where clause'

Reply:
Also, the to_s on the ID is probably unnecessary

Reply:
@jturel I think the `.to_s` may be necessary since Foreman tries the hash key as a string [here](https://github.com/theforeman/foreman/blob/62ef3e89d576611a74f132af84a67c750cb02e0d/app/models/report_composer.rb#L158)

Reply:
@sseelam2 I think this is why I'm getting the error on my end.  `TemplateInput.find_by_name("Days from Now")` finds the first `TemplateInput` with that name, but since I've been working on this feature I've actually amassed _four_ `TemplateInputs` with that name.  The one it happens to find isn't the one attached to the `@report_template`, and therefore the required input is not satisfied.

Reply:
```ruby
[14] pry(main)> TemplateInput.where(name: "Days from Now").map(&:id)
=> [15, 17, 16, 23]

[15] pry(main)> composer = ReportComposer.new({:user_id => 2, "format"=>"csv", "template_id"=>138, "input_values"=>{"15"=>{"value"=>"180"}}})
[16] pry(main)> composer.valid?
=> false

[17] pry(main)> composer = ReportComposer.new({:user_id => 2, "format"=>"csv", "template_id"=>138, "input_values"=>{"23"=>{"value"=>"180"}}})
[18] pry(main)> composer.valid?
=> true

Reply:
Need to use a where clause on the report template:

```
template_input_id = @report_template.template_inputs.find_by_name(...).id.to_s
```